src/java/io/bazel/rulesscala/scalac/ScalacWorker.java

```diff
@@ -243,6 +243,17 @@ private static String[] getPluginParamsFrom(CompileOptions ops) {
             "-P:dependency-analyzer:unused-deps-ignored-targets:"
                 + String.join(":", ignoredTargets));
       }
+    } 
+    
+    if (ops.enableSemanticDb) {
```

Comment:
@aishfenton can this decision be made in one of the phases? My goal is to not have additional logic in worker code.